### PR TITLE
Use natural sort order

### PR DESCRIPTION
**Pull request by tsipinakis**
Some feeds post multi-part articles and in that case I expect sorting by
article title to show them in order. But `strcasecmp()` does a simple
character by character comparison which sorts it like so: "1, 10, 11, 2,
3 ...".

Using `strverscmp()` changes it to the natural "1, 2, .., 10, 11"

---

There's another usage of `strcasecmp()` in `utils.cpp::convert_text`:
https://github.com/newsboat/newsboat/blob/b46c7a4bb1f282ec4471d4c1bde15f34dfe383f2/src/utils.cpp#L328

But I'm not sure what it's for so I opted to leave it alone (If it ain't broken...).

---

Note: `strverscmp` manpage says it needs `_GNU_SOURCE` defined but it appears to work without it anyway (is it defined somewhere else?).

I'm not sure if this has any implications with portability.

### CODE REVIEW

**Comment by tsipinakis** [comment]
Apparently it fails on mac :(

I couldn't find any other equivalent function in other libraries we use. So I guess we can either include an existing implementation into the source directly (like [this](http://www.davekoelle.com/alphanum.html) one I found after a bit of googling), or roll our own?

**Comment by Minoru** [comment]
I think this is a good idea, thank you!

> There's another usage of strcasecmp() in utils.cpp::convert_text

Seems okay. The purpose there is to compare user's `LC_CTYPE` environment variable with our constant (always `"utf-8"`). Locales are usually all uppercase, but I guess the check was done this way just to be safe. Here's the commit that added it: https://github.com/newsboat/newsboat/commit/864a6356bc6e4f973a9efe702827d110e8dff898#diff-4d3039dff574b788948119de402ca8d9R249

> I couldn't find any other equivalent function in other libraries we use

I don't see any such functions either. I think it's better to use an existing implementation. The one you linked to looks fine, I'd use that. Seems like we'll need to define `ALPHANUM_LOCALE`; I think this should be done right in our copy of the header, so we don't have to repeat it everywhere. You can hide the whole thing behind a function in utils (to avoid mental burden of yet another namespace), and copy tests over to our test suite.

**Comment by tsipinakis** [comment]
Done, we'll need to somehow remember to re-define `ALPHANUM_LOCALE` if we ever decide to update the header.

File: test/utils.cpp [hunk 1]
```diff
@@ -648,6 +648,34 @@ TEST_CASE("getcwd() returns current directory of the process", "[utils]")
 	}
 }
 
+TEST_CASE("strnaturalcmp() compares strings appropriately", "[utils]")
```
Comment (Minoru) [inline test/utils.cpp]: I understand that "natural" already hints at what is "appropriate" for this function, but let's be explicit and say that the function "compares strings using natural numeric ordering" or something.

**Comment by Minoru** [comment]
I apologize for dropping the ball on this one—I felt too tired for a review on Tuesday, and then just kept postponing it for no good reason.

**Comment by tsipinakis** [comment]
> I apologize for dropping the ball on this one

That's fine, noone's in a hurry :-)


I also added a small unrelated fixup commit to fix a build failure with `make test`, didn't deserve its own PR but it was annoying me!

**Comment by Minoru** [comment]
Cool! Thank you, @tsipinakis!

**Pull request merged**